File: 1325/1325.js
let fs = require("fs");
let input = fs.readFileSync("input.txt").toString().split("\n");
let [N, M] = input[0].split(" ").map(Number);
let v = Array.from({ length: N + 1 }, () => []);
let dp = new Array(N + 1).fill(-1);
let maxValue = -1;
let ans = [];

for (let i = 1; i <= M; i++) {
    let [A, B] = input[i].split(" ").map(Number);
    if (A !== B) {
        v[B].push(A);
    }
}

function bfs(node) {
    const visited = new Array(N + 1);
    let count = 1;
    visited[node] = true;
    let q = [node];
    while (q.length) {
        let cur = q.shift();
        v[cur].map((item) => {
            if (!visited[item]) {
                q.push(item);
                visited[item] = true;
                count += 1;
            }
        });
    }
    return count;
}

for (let i = 1; i <= N; i++) {
    dp[i] = bfs(i);
    maxValue = Math.max(dp[i], maxValue);
}
for (let i = 1; i <= N; i++) {
    if (maxValue === dp[i]) {
        ans.push(i);
    }
}
console.log(ans.join(" "));

File: 1325/1325.cpp
#include<iostream>
#include<algorithm>
#include<vector>
#include<cstring>
using namespace std;

int N, M;
vector<int> v[10001];
bool visited[10001];
int ans[10001];
int maxValue = -1;

int dfs(int node)
{
    visited[node] = true;
    int count = 1;
    for (auto cur : v[node])
    {
        if(!visited[cur]){
            count+=dfs(cur);
        }
    }
    return count;
}

int main()
{
    ios::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);
    cin >> N >> M;

    for (int i = 0; i < M;i++){
        int A, B;
        cin >> A >> B;
        v[B].push_back(A);
    }
    for (int i = 1; i <= N;i++){
        memset(visited, false, sizeof(visited));
        ans[i] = dfs(i);
        maxValue = max(maxValue, ans[i]);
    }
    for (int i = 1; i <= N;i++){
        if(maxValue==ans[i]){
            cout << i << " ";
        }
    }
}